src/dune_rules/cxx_flags.ml

```diff
@@ -1,16 +1,23 @@
 open! Stdune
 open Dune_engine
 
+type phase =
+  | Compile
+  | Link
+
 type ccomp_type =
   | Gcc
   | Msvc
   | Clang
   | Other of string
 
-let base_cxx_flags = function
-  | Gcc -> [ "-x"; "c++"; "-lstdc++"; "-shared-libgcc" ]
-  | Clang -> [ "-x"; "c++" ]
-  | Msvc -> [ "/TP" ]
+let base_cxx_flags ~for_ cc =
+  match (cc, for_) with
+  | Gcc, Compile -> [ "-x"; "c++" ]
+  | Gcc, Link -> [ "-lstdc++" ]
```

Comment:
I removed the flag `-shared-libgcc` since the link flags actually populates the `c_library_flags` standard fields and it is not a library.

I am wondering if it was really needed or not ? @jeremiedimino do you know the role of that flag ?

Reply:
Looking at the man page, it forces gcc to use the shared version of libgcc rather than the static one. According to David [here](https://github.com/ocaml/dune/issues/3528#issuecomment-664387257), it seems to be needed. Would be great if we had a regression test for this. Without such a test, I tend to think we should keep it by default for the reason David mentioned.

Reply:
Seems reasonable. I would like these default flags to be part of the `standard` sets of the corresponding fields. But I am unsure of were the flag `-shared-libgcc` should go. I would expect it to be part of the `c_library_flags` but according to the code this field should only contains `-l<lib>` flags as they are all prefixed by `-cclib` in `Lib_rules.build_lib` (while I guess other flags should be prefixed by `-ccopt`: 

```
let cclibs =
    Expander.expand_and_eval_set expander lib.c_library_flags ~standard
in
...
Command.run (Ok compiler) ~dir:(Path.build ctx.build_dir)
  [ ...
    ; Dyn
      (Action_builder.map cclibs ~f:(fun x ->
        Command.quote_args "-cclib" (map_cclibs x)))
```

Of couse I could change that prefixing to  use `-cclib` for `-l<lib>` flags and `-ccopt` for others. But I would like to be sure that this is the correct thing to do.

Reply:
But somewhere else these flags are wrapped with.... `-ccopt`:

```
 (Action_builder.map c_library_flags ~f:(fun cclibs ->
    (* https://github.com/ocaml/dune/issues/119 *)
    match ctx.lib_config.ccomp_type with
    | Msvc ->
       let cclibs = msvc_hack_cclibs cclibs in
       Command.quote_args "-ldopt" cclibs
                   ```

Are `-cclib` and `-ccopt` equivalent ?

Reply:
Ok I got it:
```
  -cclib <opt>  Pass option <opt> to the C linker
  -ccopt <opt>  Pass option <opt> to the C compiler and linker
  ```

I will make the required changes.